### DISCUSSION TITLE
Loading Content render fragment for data grid

### DISCUSSION THREAD

**Discussion by StephenOTT** [microsoft/fluentui-blazor#1064] [Ideas]
Similar to show a button has had a loading indicator, the data grid would benefit from it.

It is fairly simple to add a render fragment for the `Items` usage, but `ItemsProvider` usage becomes more complicated (unless i have missed something) because the variable holding the provider is a provider and is never null, so you cannot do the typical "if null then show loading indicator".

Instead you have to add an additional bool variable to track when the loading of the provider req is completed.

Is there a better way to do this?

If not and the above is the current practice, I think it would be helpful to have a loading fragment similar to EmptyContent, so we can customize the loading indicator without repetitive variable creation.

Thanks!

**Comment by mguinness** [comment]
As [FluentDataGrid](https://www.fluentui-blazor.net/DataGrid) is internally using [QuickGrid](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/quickgrid) code it might be an idea to also create an issue there using [feature-blazor-quickgrid](https://github.com/dotnet/aspnetcore/labels/feature-blazor-quickgrid) label.
